starknet_py/net/schemas/common.py

```diff
@@ -336,3 +336,17 @@ class StorageEntrySchema(Schema):
     def make_dataclass(self, data, **kwargs):
         # pylint: disable=no-self-use
         return StorageEntry(**data)
+
+
+class ChainIdField(fields.Field):
+    def _deserialize(self, value, attr, data, **kwargs):
+        if isinstance(value, int):
+            value = str(value)
+        return value
+
+
+class RevisionField(fields.Field):
+    def _deserialize(self, value, attr, data, **kwargs):
+        if isinstance(value, str):
+            value = int(value)
+        return value
```

Comment:
Revision can be both string and int, ref https://github.com/starknet-io/SNIPs/pull/79

Reply:
Yes, to be more precise - type of revision is an integer for revision 1 and a string for revision 0. Please look at [the test for revision as string](https://github.com/software-mansion/starknet-jvm/blob/7ee0e50c5291e61920ca347ad299828fcf15524a/lib/src/test/kotlin/starknet/data/TypedDataTest.kt#L806) in JVM sdk.

Reply:
Ok it makes sense 👍 Aside from that we want to return a `Revision` enum in `_deserialize`, right? Furthermore, we can verify if the value is either `0` or `1`, otherwise raise an error.

Reply:
Sure, I've added `Revision._missing_()` method with custom error message, informing about possible revision values.

Reply:
Update: after online discussion, `Revision._missing_()` has been removed. Checking invalid revision value is performed in `RevisionField._deserialize()`.